src/Components/Panels/ExploreLogsVizPanelMenu.tsx

```diff
@@ -0,0 +1,194 @@
+import { DataFrame, GrafanaTheme2, PanelMenuItem } from '@grafana/data';
+import {
+  SceneComponentProps,
+  SceneCSSGridItem,
+  sceneGraph,
+  SceneObject,
+  SceneObjectBase,
+  SceneObjectState,
+  SceneQueryRunner,
+  VizPanelMenu,
+} from '@grafana/scenes';
+import React from 'react';
+import { css } from '@emotion/css';
+import { onExploreLinkClick } from '../ServiceScene/GoToExploreButton';
+import { IndexScene } from '../IndexScene/IndexScene';
+import { getQueryRunnerFromChildren } from '../../services/scenes';
+import { reportAppInteraction, USER_EVENTS_ACTIONS, USER_EVENTS_PAGES } from '../../services/analytics';
+import { logger } from '../../services/logger';
+import { AddToExplorationButton } from '../ServiceScene/Breakdowns/AddToExplorationButton';
+import { getPluginLinkExtensions } from '@grafana/runtime';
+import { ExtensionPoints } from '../../services/extensions/links';
+
+const ADD_TO_INVESTIGATION_MENU_TEXT = 'Add to investigation';
+const ADD_TO_INVESTIGATION_MENU_DIVIDER_TEXT = 'Add to investigation div';
+
+interface ExploreLogsVizPanelMenuState extends SceneObjectState {
+  body?: VizPanelMenu;
+  frame?: DataFrame;
+  labelName?: string;
+  fieldName?: string;
+  addToExplorations?: AddToExplorationButton;
+}
+
+/**
+ * @todo the VizPanelMenu interface is overly restrictive, doesn't allow any member functions on this class, so everything is currently inlined
+ */
+export class ExploreLogsVizPanelMenu
+  extends SceneObjectBase<ExploreLogsVizPanelMenuState>
+  implements VizPanelMenu, SceneObject
+{
+  constructor(state: Partial<ExploreLogsVizPanelMenuState>) {
+    super(state);
+    this.addActivationHandler(() => {
+      this.setState({
+        addToExplorations: new AddToExplorationButton({
+          labelName: this.state.labelName,
+          fieldName: this.state.fieldName,
+          frame: this.state.frame,
+        }),
+      });
+
+      // @todo rewrite the AddToExplorationButton
+      // Manually activate scene
+      this.state.addToExplorations?.activate();
+
+      const items: PanelMenuItem[] = [
+        {
+          text: 'Explore',
+          iconClassName: 'compass',
+          shortcut: '',
+          onClick: () => onExploreClick(this),
```

Comment:
Requires some changes, but can we change this from `onClick` to `href`, i.e. change `onExploreLinkClick` to return a string instead?

That would mirror core functionality, and it would allow people to copy a link and CMD+Click on the link to topen in a new tab.

Reply:
Updated, this should work here as the links are generated on activation of the menu (when the menu is opened), but it won't work for the existing explore button

Reply:
Yep, I know, looked into that before. I'll take another look in a bit 👍